### ISSUE TITLE
How to bind path to api endpoint? 

### ISSUE THREAD

**Issue by abhinavatai** [RobertoPrevato/BlackSheep#76]
Hey, I'm trying to bind dynamic path to api endpoint in BlackSheep just like we do in flask and quart. But it's not working.

Example:

In flask and quart it can be done like this:

url: xyz.com/get/file/dynamic/path/to/file

```
@api.route('/get/file/<path:filepath>', methods=['GET'])
def get_file(filepath):
       ..... 

```

How to do this in BlackSheep?

**Comment by RobertoPrevato** [comment]
Hi @abhinavatai 
This is possible using a star, as described [here[(https://www.neoteroi.dev/blacksheep/routing/#catch-all-routes).

```python
@get("/get/file/*")
def catch_all(request):
    tail = request.route_values["tail"]
```

Thank You for opening this issue, though, because it made me realize that the code API can be improved here.

**Comment by abhinavatai** [comment]
> Hi @abhinavatai 
> This is possible using a star, as described [here](https://www.neoteroi.dev/blacksheep/routing/#catch-all-routes).
> 
> ```python
> @get("/get/file/*")
> def catch_all(request):
>     tail = request.route_values["tail"]
> ```
> 
> Thank You for opening this issue, though, because it made me realize that the code API can be improved here.

Thanks. For now I can use this. 

But as you said it can be improved here.
Do you think a new Automatic Route Binding attribute can be introduced here like **FromPath** to match the framework architecture?

**Comment by RobertoPrevato** [comment]
Yes, sure. I wouldn't call it **FromPath** for this specific case, since you are interested in the part matched by the  `*` and not the full path, but it is possible.

For example:

```python
from blacksheep.server.bindings import Binder, BoundValue


class PathStar(BoundValue[str]):
    pass


class PathStarBinder(Binder):

    handle = PathStar

    async def get_value(self, request: Request) -> Optional[str]:
        return request.route_values.get("tail")
```

Then:

```python
@get("/get/file/*")
def catch_all(tail: PathStar):
    print(tail.value)
```

However, I think I will implement support for the same kind of syntax in Flask / Quart:

```python
'/get/file/<path:filepath>'
```

Since it looks nicer and more consistent with other route parameters that can be matched by name.

**Comment by abhinavatai** [comment]
Cool. 🙌🏻
Glad that BlackSheep provides a way to write custom Route Attribute. It's really useful in these type of case. 

Looking forward to implementation of default syntax. 👍🏻

**Comment by RobertoPrevato** [comment]
Hi @abhinavatai 
This is now available in version [0.3.2](https://pypi.org/project/blacksheep/). Thanks for your constructive feedback, a nice new feauture has been added to the framework: not only it supports the same syntax like Flask, but it also supports more accurate route patterns using the same value rules, for example:

* <int:foo> will match only a web request having a path with `\d+` pattern

it is also now possible to read the path of a catch-all route using:

`<path:example>`
